integration_tests/run_pyspark_from_build.sh

```diff
@@ -117,9 +117,17 @@ else
     NUM_LOCAL_EXECS=${NUM_LOCAL_EXECS:-0}
     MB_PER_EXEC=${MB_PER_EXEC:-1024}
     CORES_PER_EXEC=${CORES_PER_EXEC:-1}
-
-    if ((NUM_LOCAL_EXECS > 0)); then
-      export PYSP_TEST_spark_master="local-cluster[$NUM_LOCAL_EXECS,$CORES_PER_EXEC,$MB_PER_EXEC]"
+    # Spark 3.1.1 includes https://github.com/apache/spark/pull/31540
+    # which helps with spurious task failures as observed in our tests. If you are running
+    # Spark versions before 3.1.1, this sets the spark.max.taskFailures to 4 to allow for
+    # more lineant configuration, else it will set them to 1 as spurious task failures are not expected
+    # for Spark 3.1.1+
+    VERSION_STRING=`$SPARK_HOME/bin/pyspark --version 2>&1|grep version\ |grep -v Scala|awk '{print $5}'`
+    echo "Detected Spark version is $VERSION_STRING"
+    IS_BEFORE_SPARK_311=`python -c "print(1 if '$VERSION_STRING' < '3.1.1' else 0)"`
```

Comment:
we can use string comparison in bash, and don't seem to need this variable
```bash
[[ "$VERSION_STRING" < "3.1.1" ]] && SPARK_TASK_MAXFAILURES=4
```

Reply:
So I mistakenly thought that python was doing something smart about versions in that compare (which is an odd assumption come to think of it). I used that because that's our impl for`is_before_spark_311()`. I think either version is technically wrong. How about:

```
python -c "from packaging import version; print(1 if version.parse('3.1.9') < version.parse('3.1.10') else 0)"
```

The above prints "1", whereas:

```
if [[ 3.1.9 < 3.1.10 ]]; then echo "1"; else echo "0"; fi
```

prints "0".

Reply:
I did not comment about this because it looked good enough to me. But this is indeed a correct way to do semantic version comparison.

Reply:
went for the simpler approach, since all we really care about is less than 3.1.1